### PR TITLE
Document withSchema and recursive types in README

### LINKED ISSUE #476: Release 4.0.0: extend reamde
The readme could cover two more advanced customization topics:
* the behavior of `withSchema` that was introduced on `Encoder` and `Decoder` to modify their behavior and allow for full customization of schemas.
* how recursive types are handled and how custom `SchemaFor`, `Encoder` and `Decoder` need to be build if they are within a recursive type.

### PR DESCRIPTION
**Pull request by andreas-schroeder**
Fixes #476

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -260,26 +260,14 @@ In order to override how a schema is generated for a particular type you need to
 To do this, we just introduce a new instance of `SchemaFor` and put it in scope when we generate the schema.
 
 ```scala
-implicit object IntOverride extends SchemaFor[Int] {
-  override def schema(fieldMapper: FieldMapper): Schema = SchemaBuilder.builder.stringType
-}
+implicit val intOverride = SchemaFor[Int](SchemaBuilder.builder.stringType)
```
Comment (sksamuel) [inline README.md:263]: intType ?

Comment (andreas-schroeder) [reply README.md:263]: do you mean intOverride -> intType, or builder.stringType -> builder.intType? If the latter, I didn't want to change the example. If the former, sure, can do (I wanted however to stay close to the original example)

Comment (andreas-schroeder) [reply README.md:263]: Hi @sksamuel, can you help me out here? I didn't get what your original comment was referring to, and I'd like to merge the readme changes into the release 4.0.x branch.

Comment (sksamuel) [reply README.md:263]: It wasn't clear to me that the override encodes an int as a string type. It was like that before, and I guess it's just to demonstrate that you can use a custom encoder to do whatever you want.

**Pull request merged**